### ISSUE TITLE
On multiple LMS assignments where instructors re-use the same PDF, only load annotations particular to that assignment

### ISSUE THREAD

**Issue by mkdir-washington-edu** [hypothesis/product-backlog#1619]
**The problem**
Instructors will end up with multiple PDFs having the same fingerprint, either because:
- They divided up the PDF improperly (or used a program that divides it improperly), leaving the same fingerprint in each sub-PDF
- They used the exact same undivided PDF over multiple assignments
- They used a pre-divided PDF which keeps the same fingerprint in all sub-PDFs (ex: LibreTexts will provide a PDF of the whole textbook or individual chapters, but all of these PDFs for a given book have the same fingerprint in them).

This causes LMS readings to end up with annotations from multiple chapters on them. By the time the instructor has noticed students have usually already annotated multiple assignments, at which point it's too late to fi for the current assignments.

**The solution**
When we see the same PDF fingerprint over different LMS assignments, we should treat each assignment as having its own distinct fingerprint. This will remove from instructors the need to check PDF fingerprints before creating assignments.

Anecdotally, some instructors have reporting purposefully using this functionality in their assignments. There may not be a way to satisfy both the use case recognizing and ignoring PDF fingerprints, but we should consider the pros and cons of a setting where instructors can choose whether or not annotations from a prior assignment appear over the current one.

**Comment by robertknight** [comment]
> Anecdotally, some instructors have reporting purposefully using this functionality in their assignments. 

Do you mean that some instructors _wanted_ the same annotations to show up across multiple assignments? Can you add more context on the use cases?

**Comment by janraev** [comment]
Another possible solution to consider is to have the option to filter annotations posted in that particular assignment only. This essentially will filter out annotations made from other assignments that are being shown due to document equivalence.

**Comment by mkdir-washington-edu** [comment]
> Do you mean that some instructors wanted the same annotations to show up across multiple assignments? Can you add more context on the use cases?

Yes, they want this. As an example, they assign annotations over chapter 1, but the PDF contains chapters 1-5. On the next assignment, students are supposed to reflect on the annotations they made on chapter 1 while annotating on chapter 2. The assignment builds from week to week that way, re-using the same PDF so that past assignment annotations remain visible.

I would expect, though, that this is a very small number of use cases. It's probably ok if we break it as long as we let our users know its coming.

On the other hand, Janrae's idea sounds better to me, depending on the level of effort it might take to implement this.

**Comment by mkdir-washington-edu** [comment]
Some customers rely on this functionality so any fix needs to allow instructor to continue having annotations appear on multiple assignments.
